app/scripts/metamask-controller.js

```diff
@@ -1851,7 +1851,9 @@ export default class MetamaskController extends EventEmitter {
    */
   setupControllerConnection(outStream) {
     const api = this.getApi()
-    const dnode = Dnode(api)
+    // the "weak: false" option is for nodejs only (eg unit tests)
+    // it is a workaround for node v12 support
+    const dnode = Dnode(api, { weak: false })
```

Comment:
This likely introduces a memory leak, right? At least that's what I thought that the built-in `weak` library was doing.

Reply:
this introduces a memory leak **in tests**, the memory leak is already present in the webapp and really needs to be fixed but not in this PR

Reply:
Oh! I hadn't realized that the browser verison of dnode doesn't use `weak`. I've just double checked this - `weak` is totally absent from the bundle.